src/microbe_stage/editor/upgrades/ChemoreceptorUpgradeGUI.cs

```diff
@@ -105,6 +105,12 @@ public void OnStartFor(OrganelleTemplate organelle, GameProperties currentGame)
                 SimulationParameters.Instance.GetCompound(Constants.CHEMORECEPTOR_DEFAULT_COMPOUND_NAME);
             var defaultConfiguration = new ChemoreceptorUpgrades(defaultCompound, null,
                 Constants.CHEMORECEPTOR_RANGE_DEFAULT, Constants.CHEMORECEPTOR_AMOUNT_DEFAULT, defaultCompound.Colour);
+
+            // Assign the default ChemoreceptorUpgrades to the OrganelleUpgrades
+            // so we can check if it's been changed at the end
+            organelle.Upgrades ??= new OrganelleUpgrades();
```

Comment:
Is this fine? This seems quite suspicious change to me as this modifies the organelle state *before* anything is accepted. Would it be possible to either compare null to be equal to the default value or just document that if the upgrade is null then applying the default values is considered a "change"?